OSDOCS-12136 OCP 4.16.15 release notes (#82664)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-16-15_{context}"]
=== RHSA-2024:7174 - {product-title} {product-version}.15 bug fix and security update

Issued: 2 October 2024

{product-title} release {product-version}.15 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:7174[RHSA-2024:7174] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:7177[RHBA-2024:7177] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.16.15 --pullspecs
----

[id="ocp-4-16-15-bug-fixes_{context}"]
==== Bug fixes

> * Previously, a local patch to kube-proxy caused OpenShift SDN to add another duplicate copy of a particular rule to the iptables ruleset every time it resynchronized. The synchronization would slow down and eventually trigger the `NodeProxyApplySlow` alert. With this release, the kube-proxy patch has been fixed and the alert no longer appears. (link:https://issues.redhat.com/browse/OCPBUGS-42159[*OCPBUGS-42159*])

Reviewer: > another duplicate copy

Is this redundant?
```suggestion
* Previously, a local patch to kube-proxy caused OpenShift SDN to add a duplicate copy of a particular rule to the iptables ruleset each time it resynchronized. The synchronization would slow down and eventually trigger the `NodeProxyApplySlow` alert. With this release, the kube-proxy patch has been fixed and the alert no longer appears. (link:https://issues.redhat.com/browse/OCPBUGS-42159[*OCPBUGS-42159*])
```

> * Previously, when the Node Tuning Operator (NTO) was configured using PerformanceProfiles it would create an ocp-tuned-one-shot systemd service. The systemd service would run prior to kubelet and blocked execution. The systemd service invokes Podman which uses an NTO image, but when the NTO image was not present Podman still tried to fetch the image and it would fail. With this release, support is added for cluster-wide proxy environment variables defined in `/etc/mco/proxy.env`. Now, Podman pulls NTO images in environments which need to use proxies for out-of-cluster connections. (link:https://issues.redhat.com/browse/OCPBUGS-42061[*OCPBUGS-42061*])

Reviewer: ```suggestion
* Previously, when the Node Tuning Operator (NTO) was configured by using PerformanceProfiles, it would create an ocp-tuned-one-shot systemd service. The systemd service would run prior to kubelet and blocked execution. The systemd service invokes Podman, which uses an NTO image. But, when the NTO image was not present, Podman still tried to fetch the image and it would fail. With this release, support is added for cluster-wide proxy environment variables defined in `/etc/mco/proxy.env`. Now, Podman pulls NTO images in environments which need to use proxies for out-of-cluster connections. (link:https://issues.redhat.com/browse/OCPBUGS-42061[*OCPBUGS-42061*])
```

> * Previously, a change in the ordering of the TextInput parameters for PatternFly v4 and v5 caused the `until` field to be improperly filled and you could not edit it. With this release, the `until` field is editable so you can input the correct information. (link:https://issues.redhat.com/browse/OCPBUGS-41996[*OCPBUGS-41996*])

Reviewer: ```suggestion
* Previously, a change in the ordering of the TextInput parameters for PatternFly v4 and v5 caused the `until` field to be improperly filled and was not editable. With this release, the `until` field is editable so you can input the correct information. (link:https://issues.redhat.com/browse/OCPBUGS-41996[*OCPBUGS-41996*])
```

* Previously, when templates were defined for each failure domain, the installation program required an external connection to download the OVA in vSphere. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41885[*OCPBUGS-41885*])

* Previously, when installing a cluster on bare metal using installer provisioned infrastructure, the installation could time out if the network to the bootstrap virtual machine is slow. With this update, the timeout duration has been increased to cover a wider range of network performance scenarios. (link:https://issues.redhat.com/browse/OCPBUGS-41845[*OCPBUGS-41845*])

> * Previously, when a hosted cluster proxy was configured and it used an identity provider (IDP) that had an http or https endpoint, the hostname of the IDP was unresolved before sending it through the proxy. Consequently, hostnames that could only be resolved by the data plane failed to resolve for IDPs. With this update, a DNS lookup is performed before sending IPD traffic through the `konnectivity` tunnel. As a result, IDPs with hostnames that can only be resolved by the data plane can be verified by the Control Plane Operator. (link:https://issues.redhat.com/browse/OCPBUGS-41372[*OCPBUGS-41372*])

Reviewer: Per ISG, host name is two words.
```suggestion
* Previously, when a hosted cluster proxy was configured and it used an identity provider (IDP) that had an http or https endpoint, the host name of the IDP was unresolved before sending it through the proxy. Consequently, host names that could only be resolved by the data plane failed to resolve for IDPs. With this update, a DNS lookup is performed before sending IPD traffic through the `konnectivity` tunnel. As a result, IDPs with host names that can only be resolved by the data plane can be verified by the Control Plane Operator. (link:https://issues.redhat.com/browse/OCPBUGS-41372[*OCPBUGS-41372*])
```

> * Previously, due to an internal bug, the Node Tuning Operator (NTO) incorrectly computed CPU masks for interrupt and network handling CPU affinity if a machine had more than 256 CPUs. This prevented proper CPU isolation on those machines and resulted in `systemd` unit failures. With this release, the NTO computes the masks correctly.(link:https://issues.redhat.com/browse/OCPBUGS-39377[*OCPBUGS-39377*])

Reviewer: This phrase _if a machine had more than 256 CPUs_ seems more important and should be earlier in the sentence.
```suggestion
* Previously, due to an internal bug, if a machine had more than 256 CPUs, the Node Tuning Operator (NTO) incorrectly computed CPU masks for interrupt and network handling CPU affinity. This prevented proper CPU isolation on those machines and resulted in `systemd` unit failures. With this release, the NTO computes the masks correctly.(link:https://issues.redhat.com/browse/OCPBUGS-39377[*OCPBUGS-39377*])
```
